arxiv/base/templates/base/header.html

```diff
@@ -15,6 +15,6 @@ <h1 class="level-item"><a href="{{ url_for('home') }}" aria-label="arxiv-logo">a
 
 <div class="container">
     <div class="user-tools is-size-7 has-text-right has-text-weight-bold" role="navigation" aria-label="User menu">
-      {% if request.session and request.session.user %}Logged in as <span class="has-text-weight-bold">{{ request.session.user.username }}</span> | <a href="{{ url_for('account') }}">My Account</a> | <a href="{{ url_for('logout') }}">Logout</a>{% else %}<a href="{{ url_for('login') }}">Login</a>{% endif %}
+      {% if request.session and request.session.user %}Logged in as <span class="has-text-weight-bold">{{ request.session.user.name.forename }} {{ request.session.user.name.surname }} {{ request.session.user.name.suffix }}</span> | <a href="{{ url_for('account') }}">My Account</a> | <a href="{{ url_for('logout') }}">Logout</a>{% else %}<a href="{{ url_for('login') }}">Login</a>{% endif %}
```

Comment:
May want to add a check for the ``name`` attribute, just to be safe. 

```suggestion
      {% if request.session and request.session.user and request.session.user.name %}Logged in as <span class="has-text-weight-bold">{{ request.session.user.name.forename }} {{ request.session.user.name.surname }} {{ request.session.user.name.suffix }}</span> | <a href="{{ url_for('account') }}">My Account</a> | <a href="{{ url_for('logout') }}">Logout</a>{% else %}<a href="{{ url_for('login') }}">Login</a>{% endif %}
```